contributed_definitions/nyaml/NXresolution.yaml

```diff
@@ -0,0 +1,42 @@
+category: base
+doc: |
+  Describes the resolution of a physical quantity.
+type: group
+NXresolution:
+  dimension:
+    doc: |
+      The physical dimension of the resolution, e.g.,
+      energy, momentum, time, etc.
+    unit: NX_ANY
+  type:
+    doc: |
+      The process by which the resolution was determined.
+    enumeration: [estimated, derived, calibrated, other]
+  note(NXnote):
+    doc: |
+      Additional details of the estimate or description of the calibration procedure
+  resolution(NXdata):
```

Comment:
I would explicitly call this response_function or instrument_response_function. For spatial resolution, it could also be used for the point spread function (2D). 
The field for the (1D) singular value of the resolution should be added at the base level of the class.

Reply:
I'm not sure whether to put a response function under `NXresolution`. For me the response function is a measure of the _intensity_ sensitivity of the whole system (or parts). I find this an important quantity as well but if we decide to have resolution and sensitivity in here I would call the base class differently. 

The point spread function I would rater call a transfer function. Though I don't know if the jargon differs between optical and electron spectroscopy!?

I would not call this `instrument...` as I could also envision this to be used as a stand-alone resolution of an instrument part, too.

Reply:
The instrument response function defines the response of a measurement system to an infinitesimally short or small input (see e.g. https://chemistry.stackexchange.com/questions/152001/spectroscopy-what-is-an-instrument-response-function)

The resolution is typically defined as FHWM of the response function.

What you mean is probably the transfer function (https://en.wikipedia.org/wiki/Transfer_function)?

Reply:
Ok in the context of time resolution `instrument_response_function` this makes sense. However, I only know this from TCSPC context (as described [here](https://www.edinst.com/instrument-response-function-widths-minimum-measurable-lifetimes-in-the-fls1000)). But is this also commonly used for other physical quantities, e.g., energy, space, momentum?

What I referred to was the [optical transfer function](https://en.wikipedia.org/wiki/Optical_transfer_function).

Reply:
I think you can define a response function for any type of resolution, that serves mathematically as convolution kernel to get from the "real" data to the measured data, and vice versa can be used for deconvolution (https://en.wikipedia.org/wiki/Deconvolution). That's my idea to allow putting it here, because if you know it well enough, you can correct for it.
There it is also called "Impulse response function".

Reply:
Ok, sounds good. Then I'll add it like this and try document it accordingly.

Reply:
I agree with Laurenz, that is a very good idea to generalize the resolution.